contrib/format-msgpack/src/main/java/org/apache/drill/exec/store/msgpack/MsgpackFormatPlugin.java

```diff
@@ -0,0 +1,212 @@
+        // "extensions"
+        // attribute
+        return DEFAULT_EXTS;
+      }
+      return extensions;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isReadBinaryAsString() {
+      return readBinaryAsString;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isLenient() {
+      return lenient;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isPrintToConsole() {
+      return printToConsole;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isLearnSchema() {
+      return learnSchema;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isUseSchema() {
+      return useSchema;
+    }
+
+    public void setUseSchema(boolean useSchema) {
+      this.useSchema = useSchema;
+    }
+
+    public void setLearnSchema(boolean learnSchema) {
+      this.learnSchema = learnSchema;
+    }
+
+    public void setExtensions(List<String> extensions) {
+      this.extensions = extensions;
+    }
+
+    public void setLenient(boolean lenient) {
+      this.lenient = lenient;
+    }
+
+    public void setReadBinaryAsString(boolean readBinaryAsString) {
+      this.readBinaryAsString = readBinaryAsString;
+    }
+
+    public void setPrintToConsole(boolean printToConsole) {
+      this.printToConsole = printToConsole;
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = 1;
+      result = prime * result + ((extensions == null) ? 0 : extensions.hashCode());
+      result = prime * result + (learnSchema ? 1231 : 1237);
+      result = prime * result + (printToConsole ? 1231 : 1237);
+      result = prime * result + (lenient ? 1231 : 1237);
+      result = prime * result + (readBinaryAsString ? 1231 : 1237);
+      result = prime * result + (useSchema ? 1231 : 1237);
+      return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (obj == null) {
+        return false;
+      }
+      if (getClass() != obj.getClass()) {
+        return false;
+      }
+      MsgpackFormatConfig other = (MsgpackFormatConfig) obj;
+      if (extensions == null) {
+        if (other.extensions != null) {
+          return false;
+        }
+      } else if (!extensions.equals(other.extensions)) {
+        return false;
+      }
+      if (learnSchema != other.learnSchema) {
+        return false;
+      }
+      if (printToConsole != other.printToConsole) {
+        return false;
+      }
+      if (readBinaryAsString != other.readBinaryAsString) {
+        return false;
+      }
+      if (lenient != other.lenient) {
+        return false;
+      }
+      if (useSchema != other.useSchema) {
+        return false;
+      }
+      return true;
+    }
+
+  }
+
+  @Override
+  public int getReaderOperatorType() {
+    return CoreOperatorType.JSON_SUB_SCAN_VALUE;
```

Comment:
This value is used to make sense of metrics. So, either set it to 0 (no information) or go ahead and add a value for your plugin.